### PR TITLE
Feature/core 1051 email invitation

### PR DESCRIPTION
**Pull request by Tschuck**
- allow contact accept after sign-up
- allow contact accept after sign-in
- allow contact accept for already logged in account
- fix everytime initially opened mnemonic dialog
- fix `sign-up` password repeat checks

### CODE REVIEW

File: core/evancore.vue.libs/src/components/login/login.ts [hunk 1]
```diff
@@ -93,33 +95,33 @@ export default class LoginComponent extends mixins(EvanComponent) {
...
-          this.form.password.value
+          this.form.password.value,
         );
       } catch (ex) {
         this.form.password.value = false;
       }
 
-      // if the password is correct, create the correct active vault in dapp-browser, so other
-      // applications can access it
+      /* if the password is correct, create the correct active vault in dapp-browser, so other
```
Comment (karladler) [inline core/evancore.vue.libs/src/components/login/login.ts]: I'm actually really not happy with this multiline comment style. Also breaks the automatic comment/uncomment functionality in vscode. Would it be possible too fix this in eslintrc?

File: dapps/onboarding.vue/src/components/accept-contact/accept-contact.ts [hunk 2]
```diff
@@ -187,7 +187,7 @@ export default class AcceptContact extends mixins(EvanComponent) {
       }, 2000);
     } catch (ex) {
       dappBrowser.utils.log(ex.message, 'error');
-      (<any> this.$refs.acceptingError).show();
+      (this.$refs.acceptingError as any).show();
```
Comment (karladler) [inline dapps/onboarding.vue/src/components/accept-contact/accept-contact.ts]: as `EvanModal` ?

File: dapps/onboarding.vue/src/components/sign-up/sign-up.ts [hunk 3]
```diff
@@ -223,13 +226,13 @@ export default class SignUp extends mixins(EvanComponent) {
...
     this.termsAccepted = new EvanFormControl('termsAccepted', false, this);
 
-    // if the user was inivted, show the welcome page
-    if (this.$route.query.inviteeAlias) {
-      this.steps.push({
-        title: '_onboarding.sign-up.welcome',
-        disabled: () => this.activeStep !== 3,
-      });
-    }
+    /* if the user was inivted, show the welcome page
```
Comment (karladler) [inline dapps/onboarding.vue/src/components/sign-up/sign-up.ts]: was the code commented intentionally? If yes remove it completely or add a comment why it needs to be kept in comment.

File: dapps/onboarding.vue/src/components/sign-up/sign-up.vue [hunk 4]
```diff
@@ -20,125 +20,143 @@
...
+          </div>
+          <div v-if="profileForm.accountType.value === 'company'">
+            <p
+              v-if="activeStep === 1"
+              class="text-center mt-3 mb-0"
+            >
+              {{ '_onboarding.sign-up.steps.company.registration.desc' | translate }}
+            </p>
+            <p
+              v-if="activeStep === 2 && activeStep !== (steps.length - 1)"
```
Comment (karladler) [inline dapps/onboarding.vue/src/components/sign-up/sign-up.vue]:
```suggestion
              v-if="activeStep === 2 && steps.length !== 3"
```

File: dapps/onboarding.vue/src/components/sign-up/sign-up.vue [hunk 5]
```diff
@@ -20,125 +20,143 @@
...
-              {{ '_onboarding.back' | translate }}
-            </evan-button>
-            <evan-button
-              :disabled="!recaptchaToken || !termsAccepted.value"
-              @click="createProfile()"
-              class="btn-block"
-              type="primary">
-              {{ '_onboarding.sign-up.create-profile.title' | translate }}
-            </evan-button>
+            <!-- don't hide them, so formular validity checks will be done correctly -->
```
Comment (karladler) [inline dapps/onboarding.vue/src/components/sign-up/sign-up.vue]:
```suggestion
            <!-- only hide them, so formular validity checks will be done correctly -->
```

File: core/evancore.vue.libs/src/components/login/login.ts (hunk 1 above)
Comment (datrinh) [reply core/evancore.vue.libs/src/components/login/login.ts]: Ofc it's possible to disable that rule. I don't mind disabling it.

File: dapps/onboarding.vue/src/components/accept-contact/accept-contact.ts (hunk 2 above)
Comment (datrinh) [reply dapps/onboarding.vue/src/components/accept-contact/accept-contact.ts]: It's old code. I don't think we should expect a full refactor for everything we touch.

**Comment by datrinh** [comment]
Approved with Karl's comments in mind.

File: core/evancore.vue.libs/src/components/login/login.ts (hunk 1 above)
Comment (Tschuck) [reply core/evancore.vue.libs/src/components/login/login.ts]: is disabled ;)

File: dapps/onboarding.vue/src/components/sign-up/sign-up.ts (hunk 3 above)
Comment (Tschuck) [reply dapps/onboarding.vue/src/components/sign-up/sign-up.ts]: Ahh i forgot to remove it, is dropped now.

**Pull request merged**